### PR TITLE
Add appveyor yaml

### PR DESCRIPTION
**Pull request by ethomson**
Note that we build with VS 2012 to ensure nice backcompat with older MSVCs and their funny restrictions.

### CODE REVIEW

File: appveyor.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+version: '{build}'
+branches:
+  only:
+  - master
+configuration: RelWithDebInfo
```
Comment (nulltoken) [inline appveyor.yaml]: I think we can safely drop this. This is a leftover of a previous attempt

File: appveyor.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    mkdir build
+    cd build
+    cmake -D ENABLE_TRACE=ON -D BUILD_CLAR=ON -D STDCALL=ON .. -G"Visual Studio 11"
+    cmake --build . --config RelWithDebInfo
+test_script:
+- ps: ctest -V .
+notifications:
+- provider: Email
+  to:
+  - emeric.fermas@gmail.com
```
Comment (nulltoken) [inline appveyor.yaml]: Shouldn't we try and configure the build result routing to IRC, CampFire, .... ? See http://www.appveyor.com/docs/notifications

**Comment by nulltoken** [comment]
FWIW, The Appveyor YAML reference can be found here http://www.appveyor.com/docs/appveyor-yml

File: appveyor.yaml (hunk 2 above)
Comment (ethomson) [inline appveyor.yaml]: You don't want all the emails going to you?

Comment (nulltoken) [inline appveyor.yaml]: I'm ok with tthat. However, I'm not sure I'll be available enough to paste them in real time in the various usual channels :stuck_out_tongue_winking_eye:

**Pull request merged**